GSA-TTS/FAC
# Revised Auditee & Auditor Certification Workflow

# Auditee & Auditor Certification Workflow

Issue: #1495 
Figma: [Auditor Cert Link](https://www.figma.com/file/lnkXJTfKzD3YBhiJZ2QFEv/FAC-final?node-id=1%3A20991&mode=dev)

## Changes:
1. The auditee and auditor certification HTML has been updated to match with the Figma docs. 
    * This includes a two-step certification form, for the checklist and signature
2. Auditee/auditor certification stores the certification objects to the SAC DB object.
    * One migration adds both fields to the DB. 
    * They have their own json schema files.
    * They are of form:
    ```
    auditor/ee_certification: {
        "auditor/ee_certification": {
            "checklist_item": True,
            "checklist_item": True,
            ...
        }
        "auditor/ee_signature:{
            "auditor/ee_name": "A Name",
            "auditor/ee_title": "A Title",
        }
    }
    ```
4. While the end data is saved to the DB, it does not autofill on the page. Autofill is handled by the session variable that the forms are stored in to bridge the gap between pages. This means that they _will_ autofill if a user reloads the page, but will _not_ autofill if the user closes the page and comes back - or does some other kind of hard-reset. Since this process is (a) short and (b) only meant to be done a single time, I think this is fine. 
5. The submission checklist _does_ update as the process is completed. But, it matters little, since it's getting a big update soon.

## How to test:
1. Switch to this branch and run normally. Make sure you apply the new migrations. You can also do a clean build with `make docker-clean` and `make docker-first-run`. 
2. Create an audit, or navigate to an existing one. 
3. Get your audit to the state that it can access `/audit/auditor-certification/{reportId}`. This can be done by completing the SF-SAC, or by manually setting the `submission_status` to `ready_for_certification` in the DB. During development, I cheated it by moving the dispatch up so that the role validation didn't kick me out. Good for local dev, bad for prod. 
4. Make sure the page works, and that the proper relics are loaded into the DB.

## Screenshots:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/a747b688-a600-4494-ad70-b86bc6b0bfd4)
![image](https://github.com/GSA-TTS/FAC/assets/91098850/809afaf0-891b-478a-ae35-c89b0085b04f)
![image](https://github.com/GSA-TTS/FAC/assets/91098850/5afa3c69-de85-4a96-a4c8-0a15810ef07e)
![image](https://github.com/GSA-TTS/FAC/assets/91098850/38ccc6a8-ce00-42ac-9c99-d0f0d0e2e16a)
